crates/ruff/src/rules/pylint/rules/else_if_used.rs

```diff
@@ -0,0 +1,45 @@
+use ruff_macros::{define_violation, derive_message_formats};
+use rustpython_parser::ast::{Stmt, StmtKind};
+
+use crate::ast::helpers::identifier_range;
+use crate::registry::Diagnostic;
+use crate::source_code::Locator;
+use crate::violation::Violation;
+
+define_violation!(
+    pub struct ElseIfUsed {}
+);
+
+impl Violation for ElseIfUsed {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Consider using \"elif\" instead of \"else\" then \"if\" to remove one indentation level")
+    }
+}
+
+/// PLR5501
+pub fn else_if_used(
+    stmt: &Stmt,
+    body: &[Stmt],
+    _orelse: &[Stmt],
+    locator: &Locator,
+) -> Option<Diagnostic> {
+    if let StmtKind::If { orelse, .. } = &stmt.node {
```

Comment:
Can we not use the `orelse` passed into the function already?

Reply:
Ah, here `stmt` is the `if` in the `else: if: ...`. I'd assumed `stmt` was the outer `if` in:

```py
if ...:
  pass
else:
  if:
    ...
```

Reply:
Yes you're right, going to give that a go.